src/Microsoft.TestPlatform.CrossPlatEngine/Execution/RunTestsWithSources.cs

```diff
@@ -62,10 +62,26 @@ protected override void BeforeRaisingTestRunComplete(bool exceptionsHitDuringRun
             if (!exceptionsHitDuringRunTests && this.executorUriVsSourceList?.Count > 0 && !this.IsCancellationRequested
                 && this.TestRunCache?.TotalExecutedTests <= 0)
             {
-                IEnumerable<string> sources = new List<string>();
-                var sourcesArray = this.adapterSourceMap.Values.Aggregate(sources, (current, enumerable) => current.Concat(enumerable)).ToArray();
-                var sourcesString = string.Join(" ", sourcesArray);
+                this.LogWarningOnNoTestsDiscovered();
+            }
+        }
 
+        private void LogWarningOnNoTestsDiscovered()
+        {
+            IEnumerable<string> sources = new List<string>();
+            var sourcesArray = this.adapterSourceMap.Values
+                .Aggregate(sources, (current, enumerable) => current.Concat(enumerable)).ToArray();
+            var sourcesString = string.Join(" ", sourcesArray);
+
+            if (this.TestExecutionContext.TestCaseFilter != null)
+            {
+                var testCaseFilterToShow = TestCaseFilterDeterminer.ShortenTestCaseFilterIfRequired(this.TestExecutionContext.TestCaseFilter);
```

Comment:
Similar to LogWarningOnNoTestsDiscoveredWithTestCaseFilter, should there be LogWarningOnNoTestsExecutedWithTestCaseFilter method?
Current  one also looks good being less lines of code in existing method. But we can do the same with discovery in that case.

Reply:
Done.